system/mp/mpicomm.cpp

```diff
@@ -49,7 +49,7 @@ class NodeCommunicator: public ICommunicator, public CInterface
 public:
     IMPLEMENT_IINTERFACE;
 
-    bool send(CMessageBuffer &mbuf, rank_t dstrank, mptag_t tag, unsigned timeout)
+    virtual bool send(CMessageBuffer &mbuf, rank_t dstrank, mptag_t tag, unsigned timeout)
```

Comment:
should also use 'override' on any virtual that's overridden.
It has often saved a lot of grief, when a base definition changes and you forget to change the derived overloaded methods, the compiler will spot it and complain when override is present.